remotespark/livyclientfactory.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2015  aggftw@gmail.com
+# Distributed under the terms of the Modified BSD License.
+
+from base64 import b64encode
+
+from log import Log
+from connectionstringutil import get_connection_string_elements
+from livysession import LivySession
+from livyclient import LivyClient
+from reliablehttpclient import ReliableHttpClient
+
+
+class LivyClientFactory(object):
```

Comment:
This is so that the classes are testable. Basically, if we move the logic to the LivyClient, we cannot have it use a dummy LivySession or make the LivySession use a dummy client.

Futhermore, it helps make the magics class testable by just passing it a dummy factory that produces mockable objects.

There's certainly other ways to achieve the same, but this was they way I chose to do it.

Do you have a certain preference for an alternative method?